documentation/tech-directives/unit-test-instructions.md

```diff
@@ -0,0 +1,43 @@
+# Introduction #
+- The purpose of creating and running Unit test is to check functionality and quality of the code from each part of the software.
+
+# Requirement #
+  - Each **core** functional class are detected by a Unit Test Class.
+  - Within Unit test class, Each core method from the detected class should be attached with **at least two** test Unit Tests     with different given input and expected result.
+  - All the Unit Tests should be **commented** to indicate the detected methods.
```

Comment:
Is this a standard? I would tend to be against over-commenting. It is often clear which function is being tested, in a unit test.

Reply:
Yea That's what I meant. The standard for testing twice for each is what my prof ask me to do in my previous assignments. We can just do once for each method.

Reply:
Sorry, I agree with the two methods for testing. What I disagree with is: 
"All the Unit Tests should be **commented** to indicate the detected methods.". I think it is often extraneous to comment tests, unless they're complex. What do you think @wrc9817 ?

Reply:
I meant that just comment on which function or method you are testing.

Reply:
I understand. I think it's extraneous to describe which method we're testing if we call the tests something descriptive, such as testBuildMyArray().

Reply:
I understand if you disagree, though.

Reply:
I'm okay with both. Or we can probably ask what aleks and katreenar think.